src/transformers/models/llama/modeling_llama.py

```diff
@@ -1022,7 +1022,11 @@ def forward(
     def _update_causal_mask(self, attention_mask, input_tensor):
         if self.config._attn_implementation == "flash_attention_2":
             causal_mask = attention_mask if (attention_mask is not None and 0 in attention_mask) else None
-            return causal_mask
+            # since the static cache is padded, you have to pass the attention mask raw.
+            # similar to https://github.com/facebookresearch/llama/commit/e9077bd24177a74aa79f406bef7d4b57fe393157
+            if input_tensor.shape[1] == 1 :
+                return None
+            return attention_mask
```

Comment:
```suggestion
            return causal_mask
```
(?) Either this change is needed, or the line `causal_mask = attention_mask if (attention_mask is not None and 0 in attention_mask) else None` above is redundant

Reply:
yep I'll fix this ! Small change but yeah